hibernate-reactive-core/src/test/java/org/hibernate/reactive/OrphanRemovalTest.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Set;
+
+import static javax.persistence.CascadeType.PERSIST;
+import static javax.persistence.CascadeType.REMOVE;
+
+
+public class OrphanRemovalTest extends BaseReactiveTest {
+
+    @Override
+    protected Configuration constructConfiguration() {
+        Configuration configuration = super.constructConfiguration();
+        configuration.addAnnotatedClass( Version.class );
+        configuration.addAnnotatedClass( Product.class );
+        configuration.addAnnotatedClass( Shop.class );
+        return configuration;
+    }
+
+    @Test
+    public void testOrphan(TestContext context) {
+        Shop shop = new Shop("shop");
+        Product product = new Product("ap1", shop);
+        product.addVersion(new Version(product));
+        shop.addProduct(product);
+        shop.addProduct(new Product("ap2", shop));
+        shop.addProduct(new Product("ap3", shop));
+        shop.addProduct(new Product("ap4", shop));
+
+        test(context,
+                getSessionFactory()
+                        .withTransaction((session, transaction) -> session.persist(shop))
+                        .thenCompose(v -> getSessionFactory().withTransaction((session, transaction) -> session.find(Shop.class, shop.id)
+                                .thenCompose( result -> session.fetch(result.products).thenApply(pl -> result) )
+                                .thenAccept(result -> {
+                                    // update
+                                    result.products.clear();
+                                    result.addProduct(new Product("bp5", result));
+                                    result.addProduct(new Product("bp6", result));
+                                    result.addProduct(new Product("bp7", result));
+                                })
+                        ))
+                        .thenCompose(v -> getSessionFactory().withTransaction((session, transaction) ->
+                                session.createQuery("select count(*) from Product", Long.class)
+                                        .getSingleResult().thenAccept(result -> context.assertEquals(3l, result))
+                        ))
+        );
+    }
+
+    @Entity(name = "ProductVersion")
+    public static class Version {
+        @Id @GeneratedValue
+        private long id;
+
+        @ManyToOne
+        private Product product;
+
+        public Version(Product product) {
+            this.product = product;
+        }
+        Version() {}
+    }
+
+    @Entity(name = "Product")
+    public static class Product {
+
+        @Id @GeneratedValue
+        private long id;
+        private String name;
+
+        Product(String name, Shop shop) {
+            this.name = name;
+            this.shop = shop;
+        }
+
+        Product() {}
+
+        @ManyToOne
+        private Shop shop;
+
+        @OneToMany(mappedBy = "product", cascade = {PERSIST, REMOVE}, orphanRemoval = true, fetch = FetchType.EAGER)
```

Comment:
So there's still a slight problem here. If I remove the `EAGER` fetch here, then removal doesn't cascade. Need to look into that further.

Reply:
Note: that's not an orphan removal; it's a cascade removal *from* the orphan.

Reply:
Alright, I know what's going on here now.

Reply:
So the original core code calls:

    entityResolver.resolve( (EntityType) types[i], value, session, entity, overridingEager )

But we call:

    ((ReactiveSession) session1).reactiveGet( entityType.getReturnedClass(), (Serializable) value )

which returns `null` when the entity has been scheduled for deletion.